### PR TITLE
Extensibility update DO NOT MERGE 

### PR DESCRIPTION
**Pull request by dkershaw10**

### CODE REVIEW

File: api-reference/beta/resources/extensionschemaproperty.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# ExtensionSchemaProperty resource type
+
+User the extensionSchemaProperty resource to define a property's name and its type, as part of a schema extension definition.
```
Comment (PatAltimore) [inline api-reference/beta/resources/extensionschemaproperty.md]: Typo. Use instead of user.

File: api-reference/beta/resources/schemaextension.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
+# schemaExtension resource type
+
+With schema extensions define a schema that you can use to extend a resource type. Once created, the schema definition is available for use on the targeted resource type.  You can then use the schema extension definition to add strongly typed custom data to a resource (either as part of creation or update of the resource). The added custom data shows up as a complex type with the unique id of the schema definition it's created from.
```
Comment (PatAltimore) [inline api-reference/beta/resources/schemaextension.md]: Awkward sentence. Consider:  Schema extensions define a schema you can use to extend a resource type.

You could drop "You can then".

The last sentence is unclear.  Does it mean?: The added custom data is a complex type containing the schema definition's unique id.

File: concepts/extensibility_open_users.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+Imagine you're building an application that is available on lots of different client platforms, such as on a desktop and mobile.  However, you want to let users configure their UI experience so it’s consistent no matter which device they use to sign in to your app. This is a pretty common problem for many apps. 
+
+For this scenario we're going to show you how to:
+
+1. Add an open extension, representing some roaming profile information about the user.
+2. Query that user and this time also get back the roaming profile.
+3. Change the roaming profile information (the open extension value).
+3. Delete an open extension value.
+
+>NOTE: This topic shows you how to add, read, update and delete open extensions on a *user* resource.  These methods are also supported for the *device, event, group, message* and *post* resource types.  So you can update the example requests below using any of those resources. The responses shown in teh examples below may be truncated for brevity.
```
Comment (PatAltimore) [inline concepts/extensibility_open_users.md]: Typos: 
So, you can..
Shown in the examples...

File: concepts/extensibility_open_users.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+1. Add an open extension, representing some roaming profile information about the user.
+2. Query that user and this time also get back the roaming profile.
+3. Change the roaming profile information (the open extension value).
+3. Delete an open extension value.
+
+>NOTE: This topic shows you how to add, read, update and delete open extensions on a *user* resource.  These methods are also supported for the *device, event, group, message* and *post* resource types.  So you can update the example requests below using any of those resources. The responses shown in teh examples below may be truncated for brevity. 
+
+## 1. Add roaming profile information
+Now imagine that the user has signs in to the app and configures the look and feel of the app.  These app settings should roam so that the user gets the same experience whatever device they sign into the app from.  Here we'll see how to add the roaming profile information to a user resource.
```
Comment (PatAltimore) [inline concepts/extensibility_open_users.md]: Typos:
Now, imagine the user signs-in to.   or signed-in to.  Later it's "sign into".   make consistent.   Generally, you use a dash as in sign-in to.

File: concepts/extensibility_open_users.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    "extensionName": "com.contoso.roamingSettings",
+    "id": "com.contoso.roamingSettings",
+    "theme": "dark",
+    "color": "purple",
+    "lang": "Japanese"
+}
+```
+
+## 2. Retrieve roaming profile information
+When the user signs in to the app from another device, the app can retrieve the signed-in user profile details as well as the roaming settings. This can be done by getting the user's resource and expanding the extension navigation property.
```
Comment (PatAltimore) [inline concepts/extensibility_open_users.md]: I'd use dashes for signs-in.

File: concepts/extensibility_open_users.md (hunk 4 above)
Comment (PatAltimore) [inline concepts/extensibility_open_users.md]: I'm wrong about this.  Jimaco corrected me and we verified with the writing guidance.  https://worldready.cloudapp.net/Styleguide/Read?id=2700&topicid=28799

Use it as a verb and don't use a dash.

File: concepts/extensibility_schema_groups.md [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+        {
+            "name": "courseType",
+            "type": "String"
+        }
+    ]
+}
+```
+
+## 3. Create a new group with extended data 
+Next we'll create a brand new group that is extended with extra data based on the *graphlearn_courses* schema extension definition that we just registered.  This is a standard ```POST``` to the *group* resource, with the additional *graphlearn_courses* resource defined in the payload.  The response will not mirror back any data extensions.  For that we need to explicitly *$select* the extension by name.
```
Comment (PatAltimore) [inline concepts/extensibility_schema_groups.md]: Typos:
Next, we'll...
For that, we...

File: concepts/extensibility_schema_groups.md [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+Content-length: 420
+{
+    "id": "dfc8016f-db97-4c47-a582-49cb8f849355",
+    "createdDateTime": "2017-02-09T00:17:05Z",
+    "description": "New Managers training course for March 2017",
+    "displayName": "New Managers March 2017",
+    "groupTypes": [
+        "Unified"
+    ],
+    "mail": "newMan201703@raphlearn.com",
```
Comment (PatAltimore) [inline concepts/extensibility_schema_groups.md]: Would it be @graphlearn.com?

File: api-reference/beta/api/schemaextension_get.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
+# Get schemaExtension
+Retrieve the properties of a specific schemaExtension resource.
+
+## Prerequisites
+The following **scopes** are required to execute this API: *Directory.Read.All* or *Directory.AccessAsUser.All*
```
Comment (dkershaw10) [inline api-reference/beta/api/schemaextension_get.md]: One of the following **scopes** is required to execute this API: *Directory.Read.All* or *Directory.AccessAsUser.All*

Yes - readwrite is overkill (as is access as user) :)

File: api-reference/beta/resources/schemaextension.md (hunk 2 above)
Comment (dkershaw10) [inline api-reference/beta/resources/schemaextension.md]: Shortened substantially to:
Schema extensions allow you to define a schema you can use to extend a resource type. Schema extension definitions let you add strongly-typed custom data to a resource. The custom data appears as a complex type on the extended resource.

File: concepts/extensibility_schema_groups.md (hunk 7 above)
Comment (dkershaw10) [inline concepts/extensibility_schema_groups.md]: Good catch!

File: api-reference/beta/resources/opentypeextension.md [hunk 9]
```diff
@@ -1,23 +1,17 @@
...
-This resource is derived from the [extension](extension.md) abstract type and has the additional **extensionName** property.
-The **extensionName** property is the only pre-defined, writable property for all extensions. One way to help make 
-sure extension names are unique is to use a reverse domain name system (DNS) method that is dependent on 
-_your own domain_, for example, `Com.Contoso.Contact`. Do not use the Microsoft domain in an extension name.
-
-
-### Use Office 365 data extensions or extended properties?
+### Use open extensions (for Outlook resources) or extended properties?
 
 Data extensions is the recommended solution for most scenarios involving storing and accessing custom data. If, however,
```
Comment (angelgolfer-ms) [inline api-reference/beta/resources/opentypeextension.md]: Do we mean "Open extensions is the recommended solution for...", in which case we should rename consistently?

File: concepts/extensibility_overview.md [hunk 10]
```diff
@@ -0,0 +1,75 @@
+# Adding custom data to Microsoft Graph (preview)
+
+Microsoft Graph provides a single API endpoint that gives you access to rich people-centric data and insights. While we’re constantly growing these data sets by adding new Microsoft cloud services, there's now a way for you to **extend** the Graph with your own custom application data. You can add custom properties to Graph resources without requiring an external data store. For example, you might decide to keep your app lightweight and store app specific user profile data in Microsoft Graph by extending the user object. Alternatively, you might want to retain your own app’s user profile store, and join this to users in the Graph by adding your app’s unique identifier to user object.
```
Comment (angelgolfer-ms) [inline concepts/extensibility_overview.md]: Please use "Microsoft Graph" consistently, not "the Graph", or "Graph" as an adjective such as "add custom properties to Graph resources".

File: concepts/extensibility_overview.md [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+
+Microsoft Graph offers 2 types of extensions. Choose the extension type that best suits your application needs:
+
+*  **Open extensions**: A good way for developers to get started with extending resources with custom data.
+*  **Schema extensions**: A more versatile mechanism for developers who care about storing typed data, making their schema discoverable and shareable, being able to filter, and in the future being able to perform input data validation and authorization.
+
+> NOTE: You should not use this feature to store sensitive personally identifiable information, such as account credentials, government identification numbers, cardholder data, financial account data, healthcare information, or sensitive background information.
+
```
Comment (angelgolfer-ms) [inline concepts/extensibility_overview.md]: I suggest we move the section "Supported resources" up to before the "Open extensions" section, just after the preamble. I think that would be one of the top questions when we introduce a new feature - which resources can one use this on?
Btw following Microsoft style guidelines, our topic titles and section titles are sentence case, not title case as in "Supported Resources". Please update for other instances as well.

File: api-reference/beta/resources/opentypeextension.md [hunk 12]
```diff
@@ -1,23 +1,17 @@
-# openTypeExtension resource type (data extensions)
+# Open extensions
```
Comment (angelgolfer-ms) [inline api-reference/beta/resources/opentypeextension.md]: Dan, because this topic is a resource reference topic, our guideline is that the topic should retain the resource name. 
That is, "openTypeExtension resource type (open extensions)" (please note the sentence case, not title case).

Laura, Vivian and I had explored the options and we decided the English-like feature name can go in parentheses, just so customers can correlate what they see in the TOC title with the topic that the TOC is linked to.

File: api-reference/beta/resources/opentypeextension.md [hunk 13]
```diff
@@ -1,23 +1,17 @@
...
 
-Office 365 data extensions are represented by the **openTypeExtension** resource. 
+Open extensions (formerly known as Office 365 data extensions) gives you an easy way to directly add untyped properties to a resource in the Microsoft Graph. Any open extension added to a resource shows up in the **extensions** navigation property, which is derived from the [extension](extension.md) abstract type.  Each extension has an additional **extensionName** property which is the only pre-defined, writable property for all extensions, along with your custom date. One way to help make sure extension names are unique is to use a reverse domain name system (DNS) method that is dependent on _your own domain_, for example, `Com.Contoso.Contact`. Do not use the Microsoft domain in an extension name.
```
Comment (angelgolfer-ms) [inline api-reference/beta/resources/opentypeextension.md]: I think we should include a similar sentence to the one you deleted. This is based on previous feedback for clarity:
"Open extensions are represented by the **openTypeExtension** resource."
For more intuitive API names (like "Outlook mail" in the TOC matching with "message resource type", that is not as confusing, but with "Open extensions" and "openTypeExtension", customers might wonder if there's a typo somewhere.

Desirably we add that back as the first sentence right after the topic title. But even if not as the first sentence in the topic, we should add it as the second sentence in this paragraph. That is:
"Open extensions (formerly known as Office 365 data extensions) gives you an easy way to directly add untyped properties to a resource in the Microsoft Graph. Open extensions are represented by the **openTypeExtension** resource. Any open extension added to a resource shows up in the **extensions** navigation property,..."

File: api-reference/beta/resources/opentypeextension.md [hunk 14]
```diff
@@ -1,23 +1,17 @@
...
-it to extend an instance of an entity type that is already defined in the Entity Data Model (EDM) - for example, a message or 
-event - by dynamically specifying custom properties and values in a JSON payload. This makes the definition of such 
-entity types more flexible, saving you time to define new entity types just for this purpose.
+Open extension example: [Add custom data to Users using Open Extensions (preview)](../../../concepts/extensibility_open_users.md)
 
-You can create data extensions of the **openTypeExtension** type in a [message](message.md), [event](event.md), or [contact](contact.md) in the signed-in user's
-mailbox, or in an **event** or [post](post.md) for an Office 365 group. In the individual-user context, the user's 
-account can be in Office 365 or a Microsoft account (Hotmail.com, Live.com, MSN.com, Outlook.com and Passport.com).
+Open extensions are supported for:
+ - a [message](message.md), [event](event.md), or [contact](contact.md)
```
Comment (angelgolfer-ms) [inline api-reference/beta/resources/opentypeextension.md]: Please add a newline before the first list item, otherwise the list is displayed as a run-on sentence.

File: api-reference/beta/resources/opentypeextension.md [hunk 15]
```diff
@@ -1,23 +1,17 @@
...
 
-You can create data extensions of the **openTypeExtension** type in a [message](message.md), [event](event.md), or [contact](contact.md) in the signed-in user's
-mailbox, or in an **event** or [post](post.md) for an Office 365 group. In the individual-user context, the user's 
-account can be in Office 365 or a Microsoft account (Hotmail.com, Live.com, MSN.com, Outlook.com and Passport.com).
+Open extensions are supported for:
+ - a [message](message.md), [event](event.md), or [contact](contact.md) 
+ - an **event** or [post](post.md) for an Office 365 group
+ - a [device](device.md)
+ - a [group](group.md)
+ - a [user](user.md)
```
Comment (angelgolfer-ms) [inline api-reference/beta/resources/opentypeextension.md]: I see that you modified organization.md for open extensions as well. Please add organization to the list above.

File: api-reference/beta/api/schemaextension_post_schemaextensions.md [hunk 16]
```diff
@@ -0,0 +1,99 @@
+# Create schemaExtension
+
+Use this API to create a new schemaExtension.
```
Comment (angelgolfer-ms) [inline api-reference/beta/api/schemaextension_post_schemaextensions.md]: Per Microsoft guideline, please provide a link to schemaExtension on first reference in a topic.

File: api-reference/beta/api/schemaextension_post_schemaextensions.md [hunk 17]
```diff
@@ -0,0 +1,99 @@
+# Create schemaExtension
```
Comment (angelgolfer-ms) [inline api-reference/beta/api/schemaextension_post_schemaextensions.md]: Please have the TOC fixed to link to schemaextension_post_schemaextensions.md. There is currently a typo in the TOC XML file, so the TOC is linked to schemaextension_post_schemaextension.md (non-existent file).

File: api-reference/beta/api/schemaextension_post_schemaextensions.md [hunk 18]
```diff
@@ -0,0 +1,99 @@
...
+
+Use this API to create a new schemaExtension.
+
+## Prerequisites
```
Comment (angelgolfer-ms) [inline api-reference/beta/api/schemaextension_post_schemaextensions.md]: Is the same Directory-based permission required to create a schema extension even for a non-directory supporting resource?

File: api-reference/beta/api/schemaextension_delete.md [hunk 19]
```diff
@@ -0,0 +1,55 @@
+# Delete schemaExtension
+
+Delete a schemaExtension.
+
+## Prerequisites
+The following **scope** is required to execute this API: *Directory.AccessAsUser.All*
```
Comment (angelgolfer-ms) [inline api-reference/beta/api/schemaextension_delete.md]: Is the same Directory-based permission required to for a non-directory supporting resource such as message?

**Comment by dkershaw10** [comment]
Yes.

File: api-reference/beta/api/schemaextension_post_schemaextensions.md (hunk 16 above)
Comment (angelgolfer-ms) [inline api-reference/beta/api/schemaextension_post_schemaextensions.md]: I am suggesting an alternative - since the Response section includes a link, and for this feature, we've got an overview concept topic in a different section of the TOC, in the summary description (first sentence) of these schema extension API topics, we should link to the "Schema extensions" section in the concept topic extensibility_overview.md. I think a link like this would be more noticeable, and connects the conceptual and reference topics for this feature better.

File: api-reference/beta/api/schemaextension_delete.md [hunk 20]
```diff
@@ -0,0 +1,55 @@
+# Delete schemaExtension
+
+Delete a schemaExtension.
```
Comment (angelgolfer-ms) [inline api-reference/beta/api/schemaextension_delete.md]: Since the schema extension feature has a set of concept topics in a different area of the TOC (not in the reference section), I suggest for the description of each of the 5 CRUD method topics, we rephrase it to use the friendly name "schema extension" and link to the "Schema extensions" section in the extensibility_overview.md topic. That would make the concept topics more discoverable and improve the SEO.
So for example, for the delete method topic, I'd go "Delete a schema extension" and link "schema extension" to the said section. Otherwise the generated, default descriptions don't add much value.

File: api-reference/beta/api/schemaextension_get.md [hunk 21]
```diff
@@ -0,0 +1,73 @@
+# Get schemaExtension
+Retrieve the properties of a specific schemaExtension resource.
```
Comment (angelgolfer-ms) [inline api-reference/beta/api/schemaextension_get.md]: I think we should modify the default, generated description to clarify this is to get the definition of the specified schema extension. And, add that to get the extended data of a resource instance such as a group, see step #3 of extensibility_schema_groups.md.
Putting real values in one or two examples in schemaextension_get.md would further help clarify as well.

File: api-reference/beta/api/schemaextension_list.md [hunk 22]
```diff
@@ -0,0 +1,78 @@
+# List schemaExtensions
+
+Retrieve a list of schemaExtension resources and their properties.
```
Comment (angelgolfer-ms) [inline api-reference/beta/api/schemaextension_list.md]: Similar to schemaextension_get.md, I suggest we clarify this description as "Get the definitions of all schema extensions."
Link the term "schema extensions" to the "Schema extensions" section in the overview topic.

Comment (angelgolfer-ms) [inline api-reference/beta/api/schemaextension_list.md]: BTW, is there a way to get all the schema extensions applicable to a resource such as group?

File: api-reference/beta/api/schemaextension_post_schemaextensions.md (hunk 16 above)
Comment (angelgolfer-ms) [inline api-reference/beta/api/schemaextension_post_schemaextensions.md]: I think we should modify this default, generated description as:
"Create and register a schema extension definition (schemaExtension object)."
Link the term "schema extension" to the "Schema extensions" section in the overview topic.
Link "schemaExtension" to the schemaextension.md resource topic.

Then, still in the preamble, we should add something to the effect of, "The targetTypes property specifies those resources that support this schema extension. After defining a schema extension, you can then create an instance of a supported resource and extend its data in the schema extension, or add a schema extension to an existing instance of a supported resource. For example, see "Create a new group with extended data" (link to #3 in the extensibility_schema_groups.md topic) and "Add an extension to an existing group" (link to #4 in that topic)."

File: api-reference/beta/api/schemaextension_post_schemaextensions.md [hunk 23]
```diff
@@ -0,0 +1,99 @@
...
+
+## Request body
+In the request body, supply a JSON representation of [schemaExtension](../resources/schemaextension.md) object.
+
+The following table shows the properties that are required when you create a schema extension.
+
+| Parameter | Type | Description|
+|:---------------|:--------|:----------|
+|description|String|Description for the schema extension.|
+|id|String|The unique identifier for the schema extension definition. The value must be a concatenation of one of your verified domains (e.g. contoso.com) and a name for the schema extension - e.g. *contoso_mySchema*. |
```
Comment (angelgolfer-ms) [inline api-reference/beta/api/schemaextension_post_schemaextensions.md]: What does "verified domain" mean?

File: api-reference/beta/api/schemaextension_post_schemaextensions.md [hunk 24]
```diff
@@ -0,0 +1,99 @@
...
+|properties|[ExtensionSchemaProperty](../resources/extensionschemaproperty.md) collection|The collection of property names and types that make up the schema extension definition.|
+|targetTypes|String collection|Set of Microsoft Graph types (that can support extensions) that the schema extension can be applied to.|
+
+## Response
+If successful, this method returns `201, Created` response code and [schemaExtension](../resources/schemaextension.md) object in the response body.
+
+## Example
+##### Request
+Here is an example of the request.
```
Comment (angelgolfer-ms) [inline api-reference/beta/api/schemaextension_post_schemaextensions.md]: It would be really helpful to provide an example with real values in the JSON request body.

File: api-reference/beta/api/schemaextension_update.md [hunk 25]
```diff
@@ -0,0 +1,98 @@
+# Update schemaextension
+
+Update the properties of a [schemaExtension](../resources/schemaextension.md) resource.
```
Comment (angelgolfer-ms) [inline api-reference/beta/api/schemaextension_update.md]: We definitely have to clarify this better - "Update the definition of a schema extension by including additional custom properties in the extension.".
Link "schema extension" to the "Schema extensions" section in the overview topic.

Then include a paragraph to describe how to update the value of a property of a schema extension (is it by calling PATCH on that resource instance?)

Comment (dkershaw10) [inline api-reference/beta/api/schemaextension_update.md]: That doesn't work.  The topic you want me to change is about updating the definition.  It's not about updating values on a resource.  That's covered in one of the Examples sections.

Comment (angelgolfer-ms) [inline api-reference/beta/api/schemaextension_update.md]: I think we're on the same page here - schemaextension_update.md is about updating the definition of a specified schema extension. That's the first paragraph of my last comment.
The second paragraph is just for the convenience of customers, like a related link. Some customers might land in this topic and realize they don't want to change the extension definition, but want to change the value of a property in the extension. We haven't explicitly provided an example for this, have we?

File: api-reference/beta/resources/schemaextension.md [hunk 26]
```diff
@@ -0,0 +1,71 @@
...
+## Properties
+| Property	   | Type	|Description|
+|:---------------|:--------|:----------|
+|description|String|Description for the schema extension.|
+|id|String|The unique identifier for the schema extension definition. The value must be a concatenation of one of your verified domains (e.g. contoso.com) and a name for the schema extension - e.g. *contoso_mySchema*. |
+|name|String|Friendly name for schema extension.|
+|owner|String|The appId of the application that created the schema extension. Read-only.|
+|properties|[ExtensionSchemaProperty](extensionschemaproperty.md) collection|The collection of property names and types that make up the schema extension definition.|
+|status|String|The lifecycle state of the schema extension. Possible states are *InDevelopment*, *Available*, and *Deprecated*. Automatically set to *InDevelopment* on creation.|
+|targetTypes|String collection|Set of Microsoft Graph types (that can support extensions) that the schema extension can be applied to. Currently this is *user, device, event, group, message, post*, with *administrativeUnit, organization* and *contact* coming soon. |
```
Comment (angelgolfer-ms) [inline api-reference/beta/resources/schemaextension.md]: Dan, FYI unlike blog posts, documentation generally doesn't include any "coming soon" and describes only current behavior. (We made one exception for delta query with the expectation that the promised behavior was certainly happening in a week or so.)

File: api-reference/beta/resources/schemaextension.md [hunk 27]
```diff
@@ -0,0 +1,71 @@
...
+
+## Lifecycle details
```
Comment (angelgolfer-ms) [inline api-reference/beta/resources/schemaextension.md]: Would you consider having this Lifecycle details section only in the concept topic (extensibility_overview.md)? Generally if there is sufficient conceptual content (like what we have) that warrants a separate conceptual topic, the reference topic (in this case schemaextension.md) should point to the conceptual topic (in this case, the "Schema extensions" section of extensibility_overview.md). 
Not repeating content can avoid getting our topics de-ranked by search engines and maintenance oversight.

File: concepts/extensibility_overview.md [hunk 28]
```diff
@@ -0,0 +1,75 @@
+# Adding custom data to Microsoft Graph (preview)
```
Comment (angelgolfer-ms) [inline concepts/extensibility_overview.md]: Dan, because open extensions have GA'd for a while and this concept topic includes open extensions, I think we should remove the"(preview)" in the TOC title and topic title. (I might have misled you on this not thinking through that your overview topic does include GA'd content.)
In the extensibility_overview.md topic, you can however append "(preview)" to the "Schema extensions" section title.

File: concepts/extensibility_overview.md [hunk 29]
```diff
@@ -0,0 +1,75 @@
...
+
+## Permissions
+The same [permission scopes](../authorization/permission_scopes.md) that are required to read from or write to a specific resource are also required to read from or write to any extensions data on that resource.  For example, for an app to be able to update the signed-in user's profile with custom app data, the app must have been granted the *User.ReadWrite.All* permission.
+
+Additionally, to create and manage schema extension definitions, an application must be granted the *Directory.AccessAsUser.All* permission.
+
+## Supported Resources
```
Comment (angelgolfer-ms) [inline concepts/extensibility_overview.md]: As mentioned in another topic, documentation generally doesn't include "coming soon" behavior, and includes only the current behavior.

File: api-reference/beta/api/schemaextension_update.md (hunk 25 above)
Comment (dkershaw10) [inline api-reference/beta/api/schemaextension_update.md]: OK - makes sense.  I can add an example of PATCH to the example topic, and then link from here.

File: api-reference/beta/resources/schemaextension.md (hunk 26 above)
Comment (dkershaw10) [inline api-reference/beta/resources/schemaextension.md]: :( OK will make the change

File: api-reference/beta/resources/schemaextension.md (hunk 27 above)
Comment (dkershaw10) [inline api-reference/beta/resources/schemaextension.md]: OK - will link.  That's a good point.

File: concepts/extensibility_overview.md (hunk 28 above)
Comment (dkershaw10) [inline concepts/extensibility_overview.md]: Hmm not sure how we line this up, because it's still preview for AAD entities.  I can make some adjustments.

File: concepts/extensibility_overview.md (hunk 29 above)
Comment (dkershaw10) [inline concepts/extensibility_overview.md]: How should we do this?  For example the 3 things with "coming soon" all ONLY support open extensions (until we make some fixes/changes).   Saying "only supported for open extensions" might suggest these will never support schema extensions...

File: concepts/extensibility_overview.md (hunk 28 above)
Comment (dkershaw10) [inline concepts/extensibility_overview.md]: Does this work?  
Under the "Open Extensions" section in the overview, I put:

>NOTE: Open extensions for administrative units, devices, groups, organization and users are only available in preview.

File: api-reference/beta/api/schemaextension_update.md (hunk 25 above)
Comment (dkershaw10) [inline api-reference/beta/api/schemaextension_update.md]: The PATCH operation for adding or updating are the same, so I'm just adding a note to the examples section.

Comment (dkershaw10) [inline api-reference/beta/api/schemaextension_update.md]: I have:  Update the properties of a schema extension definition by updating the [schemaExtension](../resources/schemaextension.md) resource.

I'm not sure how to lead in for updating a schema extension value on a resource here...

File: concepts/extensibility_overview.md (hunk 28 above)
Comment (angelgolfer-ms) [inline concepts/extensibility_overview.md]: extensions - supported resources
Hi Dan, how about we make the “Supported resources” section easier to read by using a table like  the attached?

As for the "coming soon" - how soon and certain are we talking about? I put "Coming soon" in parentheses in the picture so that we can close on that.

Comment (dkershaw10) [inline concepts/extensibility_overview.md]: OK - works for me - will make that update

File: api-reference/beta/api/schemaextension_delete.md (hunk 20 above)
Comment (angelgolfer-ms) [inline api-reference/beta/api/schemaextension_delete.md]: BTW, Dan, for delete in particular, I think it would be helpful if we add, after the method description, additional info about when one can delete a schema extension definition, and link to the "Schema extensions lifecycle" section in the overview topic for details. Looks like when a schema extension in is it Available or Deprecated state, one can't delete the extension, which might be puzzling to developers.

File: concepts/extensibility_overview.md [hunk 30]
```diff
@@ -0,0 +1,75 @@
...
+ - [message](../api-reference/beta/resources/message.md)
+ - [organization](../api-reference/beta/resources/organization.md) (coming soon  for schema extensions)
+ - [personal contact](../api-reference/beta/resources/contact.md) (coming soon  for schema extensions)
+ - [user](../api-reference/beta/resources/user.md) 
+ 
+## Known limitations for extensions
+-   Filtering on schema extension property values is not yet supported.
+-   Change tracking (Delta Query) is not supported for open or schema extension properties.
+-   Creating a resource and an open extension (using ```POST``) at the same time is not supported for administrative unit, device, group, organization or user.
+-   Deleting registered schema will remove access to data. This is a temporary limitation.
```
Comment (angelgolfer-ms) [inline concepts/extensibility_overview.md]: Oh, just noticed these few instances of "registered schema" in this section. Do we mean "a schema extension" instead?

File: api-reference/beta/api/schemaextension_update.md (hunk 25 above)
Comment (angelgolfer-ms) [inline api-reference/beta/api/schemaextension_update.md]: I have a suggestion which I'll walk through in an email shortly.

Separate from this - just so to make our reference topics more useful, would we want to include an example in the schemaextension_update.md topic to show how a developer changes the status property? Can include a link back to the "Schema extensions lifecycle" section in the overview topic. The cross-linking among topics in the same doc set would improve our ranking and SEO.

Btw, you might already have done something similar since you mentioned earlier today you would add an example about PATCH in the example concept topic. If not, this would be a good place, to improve SEO, if anything else.

File: concepts/extensibility_overview.md [hunk 31]
```diff
@@ -0,0 +1,75 @@
...
+
+> NOTE: You should not use this feature to store sensitive personally identifiable information, such as account credentials, government identification numbers, cardholder data, financial account data, healthcare information, or sensitive background information.
+
+## Open extensions
+Open extensions gives you an easy way to directly add untyped properties to a resource in the Microsoft Graph. Any open extension added to a resource shows up in the **extensions** navigation property, which is derived from the [extension](../api-reference/beta/resources/extension.md) abstract type.  Each extension has an additional **extensionName** property which is the only pre-defined, writable property for all extensions, along with your custom date. One way to help make sure extension names are unique is to use a reverse domain name system (DNS) method that is dependent on _your own domain_, for example, `Com.Contoso.Contact`. Do not use the Microsoft domain in an extension name.
```
Comment (angelgolfer-ms) [inline concepts/extensibility_overview.md]: Just noticed that currently this overview topic doesn't cross-link very well with the existing API content for open extensions and API topics for schema extensions. I'd suggest providing a link to the "Open extensions" term in the first sentence above "Open extensions gives you an easy way to..." Similarly for schema extensions - see next comment.

File: concepts/extensibility_overview.md [hunk 32]
```diff
@@ -0,0 +1,75 @@
...
+
+Open extension example: [Add custom data to Users using Open Extensions (preview)](extensibility_open_users.md)
+
+## Schema extensions
+With schema extensions you get to define the schema that you want to extend a resource class with (also known as entity-types). This allows you to extend targeted resource types with new strongly-typed properties. This schema definition must be given a unique name, based on one of your directory's verified domain names along with your extension name, for example `contoso_mySchema`.  Once created, the schema definition is available for use on the targeted resource type.  When you add custom data to a resource, it shows up as a complex type with the unique name of the schema definition it's created from. Even better, your schema, is also discoverable for other app developers to find and use, and build further experiences on top. During preview, all schemas in the "Available" state are discoverable by other app developer and may be used to extend resources with custom data as defined by the available schema extension.
```
Comment (angelgolfer-ms) [inline concepts/extensibility_overview.md]: As mentioned in the last comment, please link "schema extensions" in the first sentence above to https://msgraph-staging-devx.azurewebsites.net/en-us/docs/api-reference/beta/resources/schemaextension.htm.
